image-engine/src/main/java/com/safehill/kcrypto/image_engine/jni/UpscalingEngine.kt

```diff
@@ -0,0 +1,111 @@
+package com.safehill.kcrypto.image_engine.jni
+
+import android.graphics.Bitmap
+import kotlinx.coroutines.CoroutineScope
+import java.io.File
+import java.nio.ByteBuffer
+
+/**
+ * JNI bridge to the native UpscalingEngine implementation that can be used run inference
+ * with a pretrained model in MNN format. The model must contain a single RGB input and output
+ *
+ * @param modelFile the pretrained model to use in .mnn format
+ * @param scale expected upscaling factor of the model
+ * @param tileSize tile size to use for processing, specify 0 or negative to disable tiling
+ */
+class UpscalingEngine(val modelFile: File, val scale: Int, val tileSize: Int) {
```

Comment:
Can we rename this so it's more generic? Upscaling is confusing

Reply:
Hmm, like InferenceEngine or something similar ? What do u suggest ?

Reply:
Yep, that's better